### PR TITLE
Fix some mypy issues

### PR DESCRIPTION
**Pull request by fullonic**
Related with #83 

Solves the majority of mypy errors by:

-  Type some function args, returns and variables.
-  Add libraries stubs (if they exist).
-  Ignore the missing stubs from the libraries that don't have it. 

Due the fact that we will have a front end not written in python, the effort to reduce or eliminate mypy errors as put only in some specific directories: `./accounts/api/ `, `./jobsapp/api/` and `./tests/` or files like `models.py`.
There are some cases where `# type: ignore` is added due the complexity to match the django or django rest stubs type, but nothing critical. It can be improved later while we rewrite some parts of the backend or just ignored.

Also, because now we are using django-stubs, is necessary to add two environment variables in our CI: `SECRET_KEY` and "POSTGRES_PASSWORD` (with dummy values).

### CODE REVIEW

File: .github/workflows/code-check.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Run typing tests //will not raise
         continue-on-error: true
         run: |
-          mypy --strict $(find . | grep '.py$' | grep -ve './*/migrations')
+          mypy $(find . | grep '.py$' | grep -ve './*/migrations')
```
Comment (fullonic) [inline .github/workflows/code-check.yml:28]: This flag was moved into `.mypy.ini`

**Comment by fullonic** [comment]
I'm ignoring `tasks.py` for now because when _typed_ it creates a error when starting the app with docker:
```bash
app_1   | Traceback (most recent call last):
app_1   |   File "/usr/local/bin/invoke", line 8, in <module>
app_1   |     sys.exit(program.run())
app_1   |   File "/usr/local/lib/python3.8/site-packages/invoke/program.py", line 373, in run
app_1   |     self.parse_collection()
app_1   |   File "/usr/local/lib/python3.8/site-packages/invoke/program.py", line 465, in parse_collection
app_1   |     self.load_collection()
app_1   |   File "/usr/local/lib/python3.8/site-packages/invoke/program.py", line 696, in load_collection
app_1   |     module, parent = loader.load(coll_name)
app_1   |   File "/usr/local/lib/python3.8/site-packages/invoke/loader.py", line 76, in load
app_1   |     module = imp.load_module(name, fd, path, desc)
app_1   |   File "/usr/local/lib/python3.8/imp.py", line 234, in load_module
app_1   |     return load_source(name, filename, file)
app_1   |   File "/usr/local/lib/python3.8/imp.py", line 171, in load_source
app_1   |     module = _load(spec)
app_1   |   File "<frozen importlib._bootstrap>", line 702, in _load
app_1   |   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
app_1   |   File "<frozen importlib._bootstrap_external>", line 783, in exec_module
app_1   |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
app_1   |   File "/anem-per-feina/tasks.py", line 6, in <module>
app_1   |     def wait_for(ctx: Any, host: str, timeout: int = 30) -> None:
app_1   |   File "/usr/local/lib/python3.8/site-packages/invoke/tasks.py", line 312, in task
app_1   |     return klass(args[0], **kwargs)
app_1   |   File "/usr/local/lib/python3.8/site-packages/invoke/tasks.py", line 76, in __init__
app_1   |     self.positional = self.fill_implicit_positionals(positional)
app_1   |   File "/usr/local/lib/python3.8/site-packages/invoke/tasks.py", line 167, in fill_implicit_positionals
app_1   |     args, spec_dict = self.argspec(self.body)
app_1   |   File "/usr/local/lib/python3.8/site-packages/invoke/tasks.py", line 153, in argspec
app_1   |     spec = inspect.getargspec(func)
app_1   |   File "/usr/local/lib/python3.8/inspect.py", line 1083, in getargspec
app_1   |     raise ValueError("Function has keyword-only parameters or annotations"
app_1   | ValueError: Function has keyword-only parameters or annotations, use inspect.signature() API which can support them
```

File: tests/factories.pyi [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import factory
```
[the remark this answers is not in the document]
Comment (fullonic) [reply tests/factories.pyi:1]: It's the mypy file extension where we can provide stubs to our own modules. Ideally we would have a folder with all our stubs and give that directory to mypy to read from it.

[the remark this answers is not in the document]
Comment (francescarpi) [reply tests/factories.pyi:1]: Ah, ok! Nice!
Thank you

**Review by francescarpi — APPROVED** [review]

**Comment by francescarpi** [comment]
Hi @fullonic , so it's ok for me. Could you make a rebase with master? I'll merge it later.
Well done!

**Comment by fullonic** [comment]
@francescarpi It's already updated with master. Thanks :+1:

**Comment by francescarpi** [comment]
Thank you Diogo!!!

**Pull request merged**